oe_contact_forms.module

```diff
@@ -468,6 +473,23 @@ function _oe_contact_forms_email_validate(array &$form, FormStateInterface $form
   $form_state->setValue('corporate_fields', $values);
 }
 
+/**
+ * Validate alternative contact language field if set.
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_contact_forms_alternative_contact_language_validate(array &$form, FormStateInterface $form_state): void {
+  $values = &$form_state->getValue('corporate_fields');
+  if (empty($values['optional_fields']['oe_preferred_language'])) {
+    // Uncheck alternative contact language if preferred language is unchecked.
+    $values['optional_fields']['oe_alternative_language'] = '';
+    $form_state->setValue('corporate_fields', $values);
+  }
```

Comment:
Why we do this in a validate, we are not validating anything here we are setting the value of a field based on another field.